bruinen/src/bruinen/langchain/google.py

```diff
@@ -155,21 +161,108 @@ class GoogleGetDraftsTool(BaseTool):
     Output will be the text response from the Google API.
     """
 
+    class GoogleGetDraftsToolInputSchema(BaseModel):
+        q: Optional[str] = Field(description="The query for your drafts")
+        pageToken: Optional[str] = Field(description="The page token for your drafts")
+        
+    input_schema = GoogleGetDraftsToolInputSchema
+
     client: AuthenticatedClient
-    llm: BaseLanguageModel
     user_id: str
+    llm: Optional[BaseLanguageModel]
+    parse_parameters: Optional[Callable[[str], GoogleGetDraftsToolInputSchema]] = None
     parse_output: Optional[Callable[[GoogleDrafts], str]] = None
-
+    
+    def _parse_parameters(self, _query: str) -> GoogleGetDraftsToolInputSchema:
+        """Parse the input passed to the tool when running."""
+        parser = PydanticOutputParser(pydantic_object=self.GoogleGetDraftsToolInputSchema)
```

Comment:
Wild that the output parser doesn't handle required and null already

Reply:
I'd like to try one thing: 

Can we try writing a function (normal function), with the queries divided up:

`
def getGoogleDrafts(to: str, from:str, order: str):
  pass
`

Then we can see if this will work with openAI functions, which can handle multi-input quite easily. 

This shouldn't block the PR, just a thought on how we can simplify this.